django/contrib/gis/db/backends/mysql/operations.py

```diff
@@ -105,3 +109,15 @@ def get_geom_placeholder(self, f, value, compiler):
         else:
             placeholder = '%s(%%s)' % self.from_text
         return placeholder
+
+    def get_db_converters(self, expression):
+        converters = super(MySQLOperations, self).get_db_converters(expression)
+        if isinstance(expression.output_field, GeometryField) and self.uses_invalid_empty_geometry_collection:
+            converters.append(self.convert_invalid_empty_geometry_collection)
+        return converters
+
+    # MySQL 5.7.5, empty geometry-collection handling is improved
```

Comment:
Perhaps this could be a docstring? You might elaborate a bit more -- as someone not familiar with MySQL, it's not clear to me what "improved" means.